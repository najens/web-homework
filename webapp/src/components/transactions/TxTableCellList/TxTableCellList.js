import React from 'react'
import { bool, number, string } from 'prop-types'
import TxTableCellListItem from '../TxTableCellListItem'

export default function TxTableCellList ({
  categoryName,
  date,
  description,
  dollarAmount,
  id,
  isSelected,
  merchantName,
  userName
}) {
  const rowCells = [
    { key: 'date', val: date },
    { key: 'userName', val: userName },
    { key: 'description', val: description },
    { key: 'merchantName', val: merchantName },
    { key: 'category', val: categoryName },
    { key: 'filler', val: '' },
    { key: 'amount', val: dollarAmount.toString() }
  ]

  return rowCells.map(cell => {
    const { key, val } = cell
    return (
      <TxTableCellListItem
        columnKey={key}
        id={id}
        isSelected={isSelected}
        key={`transaction-${id}-${key}`}
        val={val}
      />
    )
  })
}

TxTableCellList.propTypes = {
  categoryName: string,
  date: string,
  description: string,
  dollarAmount: number,
  id: string,
  isSelected: bool,
  merchantName: string,
  userName: string
}
