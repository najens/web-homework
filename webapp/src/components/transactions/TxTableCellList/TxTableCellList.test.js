import React from 'react'
import { create } from 'react-test-renderer'
import TxTableCellList from './TxTableCellList'

// Only testing render since this component has no business logic
describe('TxTableCellList component', () => {
  it('renders correctly', () => {
    const testRenderer = create(
      <TxTableCellList
        categoryName='Restaurant'
        date='Mar 11'
        description='Test'
        dollarAmount='$1.00'
        id='3810c583-1307-4433-8900-8ad6d743f13d'
        merchantName='Taco Bell'
        userName='John Doe'
      />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
