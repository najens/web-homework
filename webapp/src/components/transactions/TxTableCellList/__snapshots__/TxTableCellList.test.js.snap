// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxTableCellList component renders correctly 1`] = `
Array [
  .emotion-0 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
}

<td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-date"
  >
    Mar 11
  </td>,
  .emotion-0 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
}

<td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-userName"
  >
    John Doe
  </td>,
  .emotion-0 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
}

<td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-description"
  >
    Test
  </td>,
  .emotion-0 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
}

<td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-merchantName"
  >
    Taco Bell
  </td>,
  .emotion-0 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
}

<td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-category"
  >
    Restaurant
  </td>,
  .emotion-0 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
}

<td
    className="emotion-0"
    data-testid="transaction-3810c583-1307-4433-8900-8ad6d743f13d-amount"
  >
    $1.00
  </td>,
]
`;
